cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/sql/request/SQLJoinRequest.java

```diff
@@ -19,11 +19,13 @@
 import io.cdap.cdap.api.annotation.Beta;
 import io.cdap.cdap.etl.api.join.JoinDefinition;
 
+import java.io.Serializable;
+
 /**
  * Class representing a Request to execute as join operation on a SQL engine.
  */
 @Beta
-public class SQLJoinRequest {
+public class SQLJoinRequest implements Serializable {
```

Comment:
This guy should take SQLDataSet and not dataSetName.

Reply:
This is, the datasetName is used to create a reference between a stage in Spark vs a table in the SQL engine. The Dataset name can then be used to check if a dataset exists in the SQL engine using the `exists` method, for example.

For the purpose of the engine, the SQLJoinRequest is not aware of all the SQLDatasets involved in this operation, the contract specifies that all datasets mus be pushed before this so we don't have to carry the SQLDatasets all over the application stack.

Reply:
So, I am confused on how combining would work. I believe we need a review session otherwise it takes too long in the comments

Reply:
At this stage of the implementation, we are not combining stages. Every join operation will execute as a BQ job and store results in a new table.

In a later release, when we work on the optimizations (such as combining stages or never pulling data into Spark if everything can run on BQ), we can refactor this class to add a constructor that takes in all the previous stages that will need to be combined into this one.

Reply:
I think I found a good way to implement your suggestion, let me give it a try.

Reply:
@tivv take a look.